### PR TITLE
Implement fakes to create issues

### PR DESCRIPTION
**Pull request by aroltsch**
In order to create fake issue, I need to modify the request body. Now it looks like this:

```
{ "issues": { 
	"program": { "description": "compensation", "code": "02" },
	"issue": { "description": "service connection", "code": "18" },
	"level_1": { "description": "thigh", "code": "01" },
	"level_2": { "description": "leg", "code": "07" },
	"level_3": { "description": "head", "code": "##" },
	"note": "another one"
} }
```

### CODE REVIEW

File: app/controllers/issues_controller.rb [hunk 1]
```diff
@@ -15,12 +15,12 @@ def appeal
   end
 
   def create_params
-    params.require("issues").permit(:program,
-                                    :issue,
-                                    :level_1,
-                                    :level_2,
-                                    :level_3,
-                                    :note)
+    params.require("issues").permit(:note,
```
Comment (amprokop) [inline app/controllers/issues_controller.rb]: 👍

File: app/mappers/issue_mapper.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module IssueMapper
+  def self.transform_issue_hash(issue_hash)
+    {
+      program: issue_hash[:program].try(:[], "code"),
+      issue: issue_hash[:issue].try(:[], "code"),
+      level_1: issue_hash[:level_1].try(:[], "code"),
+      level_2: issue_hash[:level_2].try(:[], "code"),
```
Comment (amprokop) [inline app/mappers/issue_mapper.rb]: for my own edification: looks like this looks for an object contained at `issue_hash[:level_2]` with a key named `"code"`, and returns that value, or `nil` if any value in the chain doesn't exist.

 the `:[]` part is throwing me, what does that mean here? thanks for explaining

Comment (aroltsch) [reply app/mappers/issue_mapper.rb]: The proper way to use `try` with a hash is with `:[]`. If I have a hash

```
h = { foo: { key: "test" }  }
=> {:foo=>{:key=>"test"}}

h[:foo].try(:[], :email)
=> nil

h[:foo].try(:key)
ArgumentError: wrong number of arguments (0 for 1)
```

Comment (aroltsch) [reply app/mappers/issue_mapper.rb]: @amprokop

**Pull request merged**